src/core/thread/context.d

```diff
@@ -0,0 +1,312 @@
+/**
+ * Every thread / fiber needs stack context data structure.
+ * thread.context decouples this info into the StackContext struct.
+ * Also, Thread / Fiber have some common functionality which is provided
+ * as a super class for both in StackContextExecutor.
+ * Finally, GlobalStackContext contains commonly used global data.
+ *
+ * Copyright: Copyright Sean Kelly 2005 - 2012.
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Authors:   Sean Kelly, Walter Bright, Alex Rønne Petersen, Martin Nowak
+ * Source:    $(DRUNTIMESRC core/thread/osthread.d)
+ */
+
+module core.thread.context;
+
+struct StackContext
+{
+    // bottom, top of stack
+    void* bstack, tstack;
+
+    /// Slot for the EH implementation to keep some state for each stack
+    /// (will be necessary for exception chaining, etc.). Opaque as far as
+    /// we are concerned here.
+    void* ehContext;
+
+    StackContext* within, next, prev;
+}
+
+/**
+A class that represents a thread of execution that manages a stack.
+This serves primarily as a superclass for Thread and Fiber.
+*/
+class StackContextExecutor
```

Comment:
part of the purpose of this class was to deduplicate as much as possible from Fiber and Thread. 

In particular:
* Thread's `m_curr` and Fiber's `m_ctxt` (this being a `StackContextExecutor` and all :) ) 
* Thread's `m_sz` and Fiber's `m_size`

are effectively the same

`m_unhandled` is identical.

There may be other ones as well.

Reply:
Ok, good. Yes, the first pass is to make the tests work and then one small change at a time. :)